app.py

```diff
@@ -40,48 +46,7 @@ def check_all_enemies_defeated():
 def generate_enemies(stage):
     global enemies
     enemies.clear()
-    
-    enemy_size_mapping = {
-        'enemy1': ENEMY1_SIZE,
-        'enemy2': ENEMY2_SIZE,
-        'boss3': BOSS3_SIZE,
-        'boss4': BOSS4_SIZE,
-        'boss5': BOSS5_SIZE,
-    }
-    
-    # Example logic for assigning health based on the stage and enemy type
-    if stage in [1, 2]:
-        enemy_count = 10
-        enemy_type = 'enemy1' if stage == 1 else 'enemy2'
-        enemy_health = enemy1_health if stage == 1 else enemy2_health
-    elif stage == 3:
-        enemy_count = 1
-        enemy_type = 'boss3'
-        enemy_health = boss3_health
-    elif stage == 4:
-        enemy_count = 1
-        enemy_type = 'boss4'
-        enemy_health = boss4_health
-    elif stage == 5:
-        enemy_count = 1
-        enemy_type = 'boss5'
-        enemy_health = boss5_health
-
-    enemy_size = enemy_size_mapping[enemy_type]
-    
-    for _ in range(enemy_count):
-        enemies.append({
-            'x': random.randint(0, WINDOW_WIDTH - enemy_size),
-            'y': 50 if stage >= 3 else random.randint(50, 150),
-            'width': enemy_size,
-            'height': enemy_size,  # Assuming square shape; adjust dimensions as necessary
-            'speed': random.choice([-1, 1]) * 2,
-            'health': enemy_health,
-            'is_boss': stage >= 3,
-            'enemy_type': enemy_type,
-            'mouth_state': 'closed',
-            'last_shot': pygame.time.get_ticks()
-        })
+    enemies = enemy_tools.get_enemies_for_stage(stage)
```

Comment:
Notice the big diff here - I took out a lot. 

Most of this logic was moved to enemy.py as well as a few other misc helper functions. It's not super perfect - i did it very quickly in like 30 minutes so don't judge me LOL - but it might give some indication as to what I meant by extracting a class earlier.